### PR TITLE
refactor: replace `chalk` and `multimatch`

### PR DESCRIPTION
**Pull request by SukkaW**
The PR replaces `chalk` w/ `picocolors` and `multimatch` w/ `picomatch`.

### CODE REVIEW

File: src/createMatcher.ts [hunk 1]
```diff
@@ -2,14 +2,15 @@
...
   // Match everything if no pattern was given
   if (typeof pattern !== 'string' && !Array.isArray(pattern)) {
     return () => true
   }
   const patterns = Array.isArray(pattern) ? pattern : [pattern]
-  return (file: string) =>
-    multimatch(path.normalize(file), patterns, { dot: true }).length > 0
+
+  const isMatch = picomatch(patterns, { dot: true })
+  return (file: string) => isMatch(path.normalize(file))
```
Comment (SukkaW) [inline src/createMatcher.ts:15]: Using `picomatch` here also helped hoisting, which prevents `patterns` from being evaluated multiple times.

File: .changeset/two-beds-collect.md [hunk 2]
```diff
@@ -0,0 +1,5 @@
+---
+'pretty-quick': patch
+---
+
+Replaces `chalk` and `multimatch` with their lightweight and performant alternative.
```
Comment (JounQin) [inline .changeset/two-beds-collect.md:5]:
```suggestion
refactor: replace `chalk` and `multimatch` with their lightweight and performant alternatives
```

**Comment by JounQin** [comment]
@SukkaW Glad to see you here again! ❤️

File: src/createMatcher.ts (hunk 1 above)
Comment (JounQin) [reply src/createMatcher.ts:15]: Will any issues be fixed by this change?

https://github.com/prettier/pretty-quick/issues?q=is%3Aissue+is%3Aopen+sort%3Aupdated-desc+pattern

Comment (SukkaW) [reply src/createMatcher.ts:15]: Doesn't know. `multimatch` is powered by the `minimatch` core. `picomatch` itself is a core that powers `nanomatch` and `micromatch`. And they do have different behaviors.

Though the existing test cases are passed, I don't know if those issues are solved. I will create another PR adding test cases covering those issues.

**Review by JounQin — APPROVED** [review]

**Comment by JounQin** [comment]
@SukkaW Do you want to

> create another PR adding test cases to cover those issues.

shortly? I'm working on prettier v3 support, and I want to wait for you first.

**Comment by SukkaW** [comment]
> @SukkaW Do you want to
> 
> > create another PR adding test cases to cover those issues.
> 
> shortly? I'm working on prettier v3 support, and I want to wait for you first.

Just finished my lunch. I will do it now.

**Comment by SukkaW** [comment]
@JounQin Done in #184.

**Pull request merged**